pkg/azure/api/api.go

```diff
@@ -59,8 +62,40 @@ type MetricsAPI interface {
 	ObserveRateLimit(operation string, duration time.Duration)
 }
 
+func constructAuthorizer(cloudName, userAssignedIdentityName string) (autorest.Authorizer, error) {
+	if userAssignedIdentityName != "" {
+		env, err := azure.EnvironmentFromName(cloudName)
+		if err != nil {
+			return nil, err
+		}
+		msiEndpoint, err := adal.GetMSIVMEndpoint()
+		if err != nil {
+			return nil, err
+		}
+
+		spToken, err := adal.NewServicePrincipalTokenFromMSIWithUserAssignedID(msiEndpoint,
+			env.ServiceManagementEndpoint,
+			userAssignedIdentityName)
+		if err != nil {
+			return nil, err
+		}
+		return autorest.NewBearerAuthorizer(spToken), nil
+	} else {
+		// Authorizer based on file first and then environment variables
+		authorizer, err := auth.NewAuthorizerFromFile(compute.DefaultBaseURI)
+		if err == nil {
+			return authorizer, nil
+		}
+		authorizer, err = auth.NewAuthorizerFromEnvironment()
```

Comment:
We can directly do a return here `return auth.NewAuthorizerFromEnvironment()`?

Reply:
That is correct, I will update the PR to make it cleaner.